src/components/HorizonalCheckboxField.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { FC } from 'react';
+import { css, cx } from 'emotion';
+import { useStyles, Checkbox, Label } from '@grafana/ui';
+import { GrafanaTheme } from '@grafana/data';
+import { useFormContext } from 'react-hook-form';
+
+const getStyles = (theme: GrafanaTheme) => ({
+  container: css`
+    display: flex;
+    align-items: center;
+    margin-bottom: ${theme?.spacing.md};
+  `,
+  checkbox: css`
+    position: relative;
+  `,
+  label: css`
+    margin-bottom: 0;
+  `,
+});
+
+interface Props {
+  disabled?: boolean;
+  id?: string;
+  name?: string;
+  label?: string;
+  description?: string;
+  className?: string;
+  value?: boolean;
+  onChange?: () => void;
+}
+
+export const HorizonalCheckboxField: FC<Props> = ({
+  disabled,
+  id,
+  name,
+  label,
+  description,
+  className,
+  value,
+  onChange,
+}) => {
+  const styles = useStyles(getStyles);
+  const { register } = useFormContext();
+
+  return (
+    <div className={cx(styles.container, className)}>
+      <Checkbox
+        name={name}
+        ref={name ? register : undefined}
+        disabled={disabled}
```

Comment:
How is this used when there is no name?

Reply:
To do some local control of a form section, like showing or hiding a form field. So in that case the checkbox value isn't part of the expected form output, I just want to use it to toggle some additional fields to include